### PR TITLE
tun/win/tunutil: Don't auto& a temporary iterator.

### PR DESCRIPTION
**Pull request by saurik**
The current Tun code fails on Windows due to trying to take a reference to a temporary:

        openvpn3/openvpn/tun/win/tunutil.hpp:213:17: error: non-const lvalue reference to type '__wrap_iter<...>' cannot bind to a temporary of type '__wrap_iter<...>'
                    for (auto& i = guids.begin(); i != guids.end(); i++)
                               ^   ~~~~~~~~~~~~~

The type of guids.begin() is a value iterator, and so needs to be declared with "auto".

### CODE REVIEW

**Comment by lstipakov** [comment]
@saurik I think commit message is a bit misleading. We only support Visual Studio on Windows and there it doesn't fail. 

Could you rephrase commit message a bit, something like "Creating non-const reference from temporary works in Visual Studio but violates standard and fails on MinGW"?

**Comment by saurik** [comment]
@lstipakov I have pushed an update that has a different commit message.

**Comment by lstipakov** [comment]
Yet another thing - have you read the CLA.rst and agree to that?  If yes, please update the commit message to include the Signed-off-by: line.

**Comment by dsommers** [comment]
Just a quick question ... the CLA stuff and all that is important to have an agreement on ... but on the patch itself .... why not use:

    for (const auto& i : guids)

instead of just changing to `auto` without reference?

**Comment by saurik** [comment]
> Yet another thing - have you read the CLA.rst and agree to that? If yes, please update the commit message to include the Signed-off-by: line.

I agree to the CLA and have pushed a new commit with --signoff.

> Just a quick question ... the CLA stuff and all that is important to have an agreement on ... but on the patch itself .... why not use:

Iterators are fundamentally a value type and should be annotated as such; regardless, C++ offers a strange feature wherein if you capture a temporary to a const & with auto storage (not auto type inference, but "on the stack") it will extend the lifetime of the temporary to the scope of the reference; but, you cannot use this here because an iterator is only useful if it is mutable: the i++ at the end of this same line of code would then not compile because we are trying to call a non-const member function on a const reference.

**Comment by lstipakov** [comment]
Yeah you cannot use const iterator with begin/end semantics but in this example `for (const auto& i : guids)` you do not have increment and it also looks more succinct. 

But anyway, I am good with current patch. @dsommers ?

**Comment by saurik** [comment]
Oh, sorry: I somehow failed to notice you also were switching to the newer for syntax in that example. I didn't want to make arbitrary other changes to the code than the minimal change required, as I didn't want to break the build on any other compilers and don't know what features you consider "ok" to use in the codebase. Regardless: yes, if it were my project, I would use the syntax you suggest (and would encourage a later patch from a project maintainer to update that consistently across the codebase and preferably then ban the explicit syntax going forward).

**Comment by dsommers** [comment]
This has been merged as commit 1406187bfc2123d29031defa5dc605946ecc433a
